src/sys/socket/addr.rs

```diff
@@ -330,6 +330,11 @@ impl InetAddr {
         match *std {
             net::SocketAddr::V4(ref addr) => {
                 InetAddr::V4(libc::sockaddr_in {
+                    #[cfg(any(target_os = "dragonfly", target_os = "freebsd",
+                              target_os = "haiku", target_os = "hermit",
```

Comment:
I pulled the list of platforms which define these from rust-lang/libc. Although nix doesn't support Haiku or Hermit, there are other `cfg` blocks which reference them in nix, so I included them here.

I omitted newlib and VxWorks, as nix doesn't currently have any `cfg` blocks for them.